### PR TITLE
Allow skipping generated files

### PR DESCRIPTION
**Pull request by avorima**
Add a command-line flag that, if enabled, skips processing auto-generated files. The auto-generated files are recognized by certain comments like `Code generated by ...` or `DO NOT EDIT`.
This feature is only implemented for the CLI, since the analyzer used in `golangci-lint` will already receive a subset of files that are not auto-generated.

Right now it's disabled by default, but that is just to not introduce a breaking change. I think it should be enabled by
default. This might require the flag to be "inverted", e.g. renamed to `IncludeGeneratedFiles` or similar.

The testdata and the function that recognizes auto-generated files were taken almost verbatim from https://github.com/golangci/golangci-lint.

Fixes: https://github.com/daixiang0/gci/issues/66

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Flags:
                                    Std | Standard - Captures all standard packages if they do not match another section
                                     (default [Standard,Default])
   -x, --SectionSeparator strings   SectionSeparators are inserted between Sections (default [NewLine])
+      --SkipGeneratedFiles         Don't process generated files
```
Comment (daixiang0) [inline README.md:41]: I recommend that add string array as params here to make users easier do custom skip.

File: pkg/gci/gci.go [hunk 2]
```diff
@@ -141,3 +145,25 @@ func LoadFormatGoFile(file io.FileObj, cfg GciConfiguration) (unmodifiedFile, fo
...
+// Taken from https://github.com/golangci/golangci-lint.
+func isGeneratedFileByComment(doc string) bool {
+	const (
+		genCodeGenerated = "code generated"
+		genDoNotEdit     = "do not edit"
+		genAutoFile      = "autogenerated file" // easyjson
+	)
+
+	markers := []string{genCodeGenerated, genDoNotEdit, genAutoFile}
+	doc = strings.ToLower(doc)
```
Comment (daixiang0) [inline pkg/gci/gci.go:161]: I would keep case sensitive.

Comment (avorima) [reply pkg/gci/gci.go:161]: This is taken directly from golangci-lint and I think it's fine. Not every code generation tool does it the same way.

**Comment by daixiang0** [comment]
Please fix CI then we can make it land.

**Pull request merged**